LiveSched/src/main/java/dev/coms4156/project/livesched/Task.java

```diff
@@ -0,0 +1,117 @@
+package dev.coms4156.project.livesched;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+/**
+ * Represents a task that has to be done. 
+ * This class stores the id, resources needed, and the location of the task.
+ */
+public class Task {
+  private final String taskId;
+  private Map<ResourceType, Integer> resourceList; // Key = ResoureceType, Value = Units needed
+  private int priority; 
+  private LocalDateTime startTime;
+  private LocalDateTime endTime;
+  private double latitude;
+  private double longitude;
+
+  /**
+   * Constructs a new Task object with the given parameters.
+   *
+   * @param taskId        the unique ID of the resource
+   * @param resourceList  the map of resources and their quantity needed
+   * @param priority      the priority of the task
+   * @param endTime       the time that task ends
+   * @param latitude      the latitude of the resource's location
+   * @param longitude     the longitude of the task's location
+   */
+  public Task(String taskId, Map<ResourceType, Integer> resourceList, int priority, 
+        LocalDateTime endTime, double latitude, double longitude) {
+    this.taskId = taskId;
+    this.resourceList = resourceList;
+    this.priority = priority;
+    this.startTime = LocalDateTime.now(); // Initially starting now
```

Comment:
I think it makes sense to set the startTime as now in the case of ERs, but I thought we should try to make our service more general and should allow future scheduling of tasks, so how about we accept the startTime as one of the parameters?

Reply:
True. I think that's a good question to ask. Maybe currentTime would be more appropriate? Just so that we can query the availability of Resource given the current time. I'm not sure.